File: C/AMS5812.c
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// AMS5812
// This code is designed to work with the AMS5812_I2CS_1000-D I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/content/Pressure?sku=AMS5812_I2CS_1000-D#tabs-0-product_tabset-2

#include <stdio.h>
#include <stdlib.h>
#include <linux/i2c-dev.h>
#include <sys/ioctl.h>
#include <fcntl.h>

void main()
{
	// Create I2C bus
	int file;
	char *bus = "/dev/i2c-1";
	if((file = open(bus, O_RDWR)) < 0)
	{
		printf("Failed to open the bus. \n");
		exit(1);
	}
	// Get I2C device, AMS5812 I2C address is 0x78(120)
	ioctl(file, I2C_SLAVE, 0x78);

	// Read 4 bytes of data
	// pressure msb, pressure lsb, temp msb, temp lsb
	char data[4] = {0};
	if(read(file, data, 4) != 4)
	{
		printf("Error : Input/Output error \n");
	}
	else 
	{
		// Convert the data
		float pressure = (data[0] * 256 + data[1]);
		float temp = (data[2] * 256 + data[3]);

		pressure = ((pressure - 3277.0) / ((26214.0) / 100.0)) - 100.0;
		float cTemp = ((temp - 3277.0) / ((26214.0) / 110.0)) - 25.0;
		float fTemp = (cTemp * 1.8 ) + 32;

		// Output data to screen
		printf("Pressure is : %.2f PSI\n", pressure);
		printf("Temperature in Celsius : %.2f C\n", cTemp);
		printf("Temperature in Fahrenheit : %.2f F\n", fTemp);
	}
}

File: Java/AMS5812.java
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// AMS5812
// This code is designed to work with the AMS5812_I2CS_1000-D I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/content/Pressure?sku=AMS5812_I2CS_1000-D#tabs-0-product_tabset-2

import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class AMS5812
{
	public static void main(String args[]) throws Exception
	{
		// Create I2C bus
		I2CBus Bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, AMS5812 I2C address is 0x78(120)
		I2CDevice device = Bus.getDevice(0x78);
		
		// Read 4 bytes of data
		// pressure msb, pressure lsb, temp msb, temp lsb
		byte[] data = new byte[4];
		device.read(data, 0, 4);
		
		// Convert the data
		double pressure = ((data[0] & 0xFF)* 256 + (data[1] & 0xFF));
		double temp = ((data[2] & 0xFF)* 256 + (data[3] & 0xFF));
		
		pressure = ((pressure - 3277.0) / ((26214.0) / 100.0)) - 100.0;
		double cTemp = ((temp - 3277.0) / ((26214.0) / 110.0)) - 25.0;
		double fTemp = (cTemp * 1.8 ) + 32;
		
		// Output data to screen
		System.out.printf("Pressure is : %.2f PSI%n", pressure);
		System.out.printf("Temperature in Celsius : %.2f C%n", cTemp);
		System.out.printf("Temperature in Fahrenheit : %.2f F%n", fTemp);
	}
}
